src/pydargs/__init__.py

```diff
@@ -45,11 +45,32 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, **kwargs: Any)
         An instance of tp.
     """
     namespace = _create_parser(tp, **kwargs).parse_args(args)
-    return tp(**namespace.__dict__)
+    return _create_object(tp, namespace)
+
+
+def _create_object(tp: Type[Dataclass], namespace: Namespace, prefix: str = "") -> Dataclass:
+    for field in fields(tp):
+        if hasattr(field.type, "__dataclass_fields__"):
+            # Create nested dataclass object
+            setattr(
+                namespace,
+                prefix + field.name,
+                _create_object(field.type, namespace, prefix=f"{prefix}{field.name}_"),
+            )
+    args = {key[len(prefix) :]: value for key, value in namespace.__dict__.items() if key.startswith(prefix)}
+    for key in args.keys():  # Remove used keys from the namespace
```

Comment:
What do you mean by "used keys"?

Reply:
The keys that have been used to instantiate an object.

For example, in the example at the end of the readme, the (dict repr of the) namespace could look like:
`{"verbose": "yes", "config_field_a": 12, "config_field_b": "def"}`.

 First `Config` is instantiated using all keys that start with `"config_"` (and with the prefix removed), and added to the namespace:
 `config=Config(field_a=12, field_b="def")`.

Then we instantiate `Base`, with all keys that start with `""`, i.e. all keys. This requires that `config_field_a` and `config_field_b` have been removed.